ingest/usecase/plugins/orderbookclaimer/ordebook_filler_ingest_plugin.go

```diff
@@ -0,0 +1,166 @@
+package orderbookclaimer
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/osmosis-labs/osmosis/osmomath"
+	"github.com/osmosis-labs/sqs/domain"
+	"github.com/osmosis-labs/sqs/domain/keyring"
+	"github.com/osmosis-labs/sqs/domain/mvc"
+	orderbookdomain "github.com/osmosis-labs/sqs/domain/orderbook"
+	orderbookgrpcclientdomain "github.com/osmosis-labs/sqs/domain/orderbook/grpcclient"
+	orderbookplugindomain "github.com/osmosis-labs/sqs/domain/orderbook/plugin"
+	passthroughdomain "github.com/osmosis-labs/sqs/domain/passthrough"
+	"github.com/osmosis-labs/sqs/log"
+	"github.com/osmosis-labs/sqs/slices"
+	"go.opentelemetry.io/otel"
+	"go.uber.org/zap"
+)
+
+// orderbookClaimerIngestPlugin is a plugin that fills the orderbook orders at the end of the block.
+type orderbookClaimerIngestPlugin struct {
+	keyring             keyring.Keyring
+	poolsUseCase        mvc.PoolsUsecase
+	orderbookusecase    mvc.OrderBookUsecase
+	orderbookRepository orderbookdomain.OrderBookRepository
+	orderBookClient     orderbookgrpcclientdomain.OrderBookClient
+
+	atomicBool atomic.Bool
+
+	logger log.Logger
+}
+
+var _ domain.EndBlockProcessPlugin = &orderbookClaimerIngestPlugin{}
+
+const (
+	tracerName = "sqs-orderbook-claimer"
+)
+
+var (
+	tracer = otel.Tracer(tracerName)
+)
+
+func New(
+	keyring keyring.Keyring,
+	orderbookusecase mvc.OrderBookUsecase,
+	poolsUseCase mvc.PoolsUsecase,
+	orderbookRepository orderbookdomain.OrderBookRepository,
+	orderBookClient orderbookgrpcclientdomain.OrderBookClient,
+	passthroughGRPCClient passthroughdomain.PassthroughGRPCClient,
+	orderBookCWAPIClient orderbookplugindomain.OrderbookCWAPIClient,
+	logger log.Logger,
+) *orderbookClaimerIngestPlugin {
+	return &orderbookClaimerIngestPlugin{
+		keyring:             keyring,
+		orderbookusecase:    orderbookusecase,
+		orderbookRepository: orderbookRepository,
+		orderBookClient:     orderBookClient,
+		poolsUseCase:        poolsUseCase,
+
+		atomicBool: atomic.Bool{},
+
+		logger: logger,
+	}
+}
+
+// ProcessEndBlock implements domain.EndBlockProcessPlugin.
+func (o *orderbookClaimerIngestPlugin) ProcessEndBlock(ctx context.Context, blockHeight uint64, metadata domain.BlockPoolMetadata) error {
+	ctx, span := tracer.Start(ctx, "orderbooktFillerIngestPlugin.ProcessEndBlock")
+	defer span.End()
+
+	orderbooks, err := o.poolsUseCase.GetAllCanonicalOrderbookPoolIDs()
+	if err != nil {
+		o.logger.Error("failed to get all canonical orderbook pool IDs", zap.Error(err))
+		return err
+	}
```

Comment:
The metadata contains the pools that were modified within a block. We should only process the intersection of modified pools from the `metadata` and `orderbooks`.

Otherwise, we would be doing redundant processing.